core/domain/state_domain.py

```diff
@@ -745,16 +758,336 @@ def get_all_html_content_strings(self):
+                schema: dict. The customization dict to be modified: dict
+                    with a single key, 'value', whose corresponding value is the
+                    value of the customization arg.
+
+            Returns:
+                dict. The converted customization dict.
+            """
+            schema_type = schema['type']
+            schema_obj_type = schema.get('obj_type', None)
+
+            if schema_obj_type == schema_utils.SCHEMA_OBJ_TYPE_SUBTITLED_HTML:
+                value['html'] = conversion_fn(value['html'])
+            elif schema_type == schema_utils.SCHEMA_TYPE_LIST:
+                for i in python_utils.RANGE(len(value)):
+                    value[i] = traverse_ca_by_schema_and_convert_html(
+                        value[i],
+                        schema['items'])
+            elif schema_type == schema_utils.SCHEMA_TYPE_DICT:
+                for property_spec in schema['properties']:
+                    name = property_spec['name']
+                    value[name] = (
+                        traverse_ca_by_schema_and_convert_html(
+                            value[name],
+                            property_spec['schema'])
+                    )
+
+            return value
+
+        ca_specs = interaction_registry.Registry.get_interaction_by_id(
+            interaction_id).customization_arg_specs
+        customization_args = interaction_dict['customization_args']
+        for ca_spec in ca_specs:
+            ca_spec_name = ca_spec.name
+            if ca_spec_name in customization_args:
+                customization_args[ca_spec_name]['value'] = (
+                    traverse_ca_by_schema_and_convert_html(
+                        customization_args[ca_spec_name]['value'],
+                        ca_spec.to_dict()['schema']
+                    )
+                )
+
+        return interaction_dict
+
+    @staticmethod
+    def convert_customization_args_dict_to_customization_args(
+            interaction_id, customization_args_dict):
+        """Converts customization arguments dictionary to customization
+        arguments. This is done by converting each customization argument to a
+        InteractionCustomizationArg domain object.
+
+        Args:
+            interaction_id: str. The interaction id.
+            customization_args_dict: dict. A dictionary of customization
+                argument name to a customization argument dict, which is a dict
+                of the single key 'value' to the value of the customization
+                argument.
+
+        Returns:
+            dict. A dictionary of customization argument names to the
+            InteractionCustomizationArg domain object's.
+        """
+        if interaction_id is None:
+            return {}
+
+        customization_args = {}
+        ca_specs = interaction_registry.Registry.get_interaction_by_id(
+            interaction_id).customization_arg_specs
+
+        for ca_spec in ca_specs:
```

Comment:
Perhaps dict comprehension might simplify this too (and you won't need line 857).